pkg/informer/informer.go

```diff
@@ -58,14 +59,64 @@ const (
 	byGroupVersionResourceIndex = "byGroupVersionResource"
 )
 
-// DynamicDiscoverySharedInformerFactory is a SharedInformerFactory that
+// GVRPartialMetadata provides the required metadata about a GVR
+// for which an informer should be started.
+type GVRPartialMetadata struct {
+	Names apiextensionsv1.CustomResourceDefinitionNames
+	Scope apiextensionsv1.ResourceScope
+}
+
+// GVRSource is a source of information about available GVRs,
+// and provides a way to register for changes in the available GVRs (addition or removal)
+// so that the list of informers can be updated.
+type GVRSource interface {
```

Comment:
Is this more than 
```
type Resources struct {
  discovery.ServerResourcesInterface
  Changes() <- chan struct{}
}
```
?

Introducing yet another discovery concept feels wrong.